lib/Model/Importer/Binary/ELFImporter.cpp

```diff
@@ -445,7 +445,9 @@ void ELFImporter<T, HasAddend>::findMissingTypes(object::ELFFile<T> &TheELF,
   unsigned MaximumRecursionDepth = 1;
 
   LDDTree Dependencies;
-  lddtree(Dependencies, InputPath, MaximumRecursionDepth);
+  std::string BinaryPath = InputPath.empty() ? TheBinary.getFileName().str() :
+                                               InputPath;
```

Comment:
1. Use `const std::string &` if you can, save one copy.
2. We also need to do this for `PDBImporter`.

Reply:
1. `getFileName()` returns a `StringRef`, so `.str()` is a temporary
2. `PDBImporter` does not have an equivalent instance, there's only [this use](https://github.com/revng/revng/blob/develop/lib/Model/Importer/DebugInfo/PDBImporter.cpp#L490) of `InputPath` and it's already guarded

Reply:
1. `const std::string &` works well with temporaries. It extends their lifetime.
2. Mmmh, yeah, checking again that's fine. We'll eventually need to get rid of this.

Reply:
1. Fixed